File: 202-happy-number/happy-number.py
class Solution:

    def sumofSquareOfDigits(self,n):
        total =0

        while n>0:
            dig=n%10
            total =total+(dig*dig)
            n=n//10
        return total

    def isHappy(self, n: int) -> bool:

        slow=n
        fast=n

        while fast!=1:
           slow =self.sumofSquareOfDigits(slow)

           fast=self.sumofSquareOfDigits(self.sumofSquareOfDigits(fast))
           if fast ==1:
              return True
           if slow==fast:
              return False
        return True   
        

File: 202-happy-number/happy-number.cpp
class Solution {
public:

   int sumofSquareofDigits(int n){
    int sum =0;

    while(n>0){
        int dig = n%10;
        sum = sum+(dig*dig);
        n = n/10;
    }
    return sum;
   }
    bool isHappy(int n) {
        int slow=n;
        int fast = n;

        while(fast!=1){
            //slow move 1 
            slow = sumofSquareofDigits(slow);
            //Fast move 2
            fast = sumofSquareofDigits(sumofSquareofDigits(fast));{
        }
        if(fast ==1){
            return true;
        }
        if(slow == fast){
            return false;
        }
        }
        return true;
    }
};